include/dem/particle_particle_contact_force.h

```diff
@@ -1087,6 +1087,164 @@ class ParticleParticleContactForce
+   * @param tangential_force Contact tangential force
+   * @param particle_one_tangential_torque Contact tangential torque on particle one
+   * @param particle_two_tangential_torque Contact tangential torque on particle two
+   * @param rolling_resistance_torque Contact rolling resistance torque
+   */
+  inline void
+  calculate_DMT_contact(particle_particle_contact_info<dim> &contact_info,
+                        const Tensor<1, 3> &tangential_relative_velocity,
+                        const double        normal_relative_velocity_value,
+                        const Tensor<1, 3> &normal_unit_vector,
+                        const double        normal_overlap,
+                        const ArrayView<const double> &particle_one_properties,
+                        const ArrayView<const double> &particle_two_properties,
+                        Tensor<1, 3>                  &normal_force,
+                        Tensor<1, 3>                  &tangential_force,
+                        Tensor<1, 3> &particle_one_tangential_torque,
+                        Tensor<1, 3> &particle_two_tangential_torque,
+                        Tensor<1, 3> &rolling_resistance_torque)
+  {
+    // Calculation of effective radius and mass
+    this->find_effective_radius_and_mass(particle_one_properties,
+                                         particle_two_properties);
+
+    const unsigned int particle_one_type =
+      particle_one_properties[PropertiesIndex::type];
+    const unsigned int particle_two_type =
+      particle_two_properties[PropertiesIndex::type];
+
+    const double radius_times_overlap_sqrt =
+      sqrt(this->effective_radius * normal_overlap);
+    const double model_parameter_sn =
+      2.0 *
+      this->effective_youngs_modulus[vec_particle_type_index(
+        particle_one_type, particle_two_type)] *
+      radius_times_overlap_sqrt;
+    double model_parameter_st =
+      8.0 *
+      this->effective_shear_modulus[vec_particle_type_index(
+        particle_one_type, particle_two_type)] *
+      radius_times_overlap_sqrt;
+
+    // Calculation of normal and tangential spring and dashpot constants
+    // using particle properties
+    double normal_spring_constant = 0.66665 * model_parameter_sn;
+    double normal_damping_constant =
+      -1.8257 *
+      this->model_parameter_beta[vec_particle_type_index(particle_one_type,
+                                                         particle_two_type)] *
+      sqrt(model_parameter_sn * this->effective_mass);
+    double tangential_spring_constant =
+      8.0 *
+        this->effective_shear_modulus[vec_particle_type_index(
+          particle_one_type, particle_two_type)] *
+        radius_times_overlap_sqrt +
+      DBL_MIN;
+    double tangential_damping_constant =
+      normal_damping_constant * sqrt(model_parameter_st / model_parameter_sn);
+
+    // Calculation of normal force. The cohesive term is computed here.
+    const double cohesive_term =
```

Comment:
Not coherent with the documentation. I think the implementation is right because of the units.

Reply:
Can you be more precise. What is not coherent?

Reply:
ahh ok, understood